File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class CasillaJuez < Casilla
  def initialize(nombre,carcel)
    super(nombre)
    @carcel = carcel
  end
  
  def recibeJugador(actual,todos)
    if(jugadorCorrecto(actual,todos))
      super(actual,todos)
      todos[actual].encarcelar(@carcel)
    end
  end
  
  def toString
    cadena="\nCasilla: #{@nombre} \n\n"
    return cadena
  end
  public :toString

end
end

File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author pc
 */
import java.util.ArrayList;

public class CasillaJuez extends Casilla {
    private static int carcel;
    
    CasillaJuez(int numCasillaCarcel,String nombre){
            super(nombre);
            carcel=numCasillaCarcel;
    }  
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
             informe(actual,todos);
             todos.get(actual).encarcelar(carcel);
        }
           
    }
    @Override
    public String toString() {
        return "CasillaJuez \n      Nombre:" + this.getNombre();
    }
    
}
